packages/core/src/browser/label-provider.ts

```diff
@@ -48,6 +49,11 @@ export interface LabelProviderContribution {
      */
     getLongName?(element: object): string;
 
+    /**
+     * Emit when something has changed that may result in this label provider returning a different
+     * value for one or more properties (name, icon etc) for the emitted URI.
+     */
+    readonly onElementUpdated?: Event<URI>;
```

Comment:
Should not it be aligned with rest of API and support other objects, not only URIs?

btw we usually call it `onDidChange`, see https://github.com/theia-ide/theia/wiki/Coding-Guidelines#event_names